src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java

```diff
@@ -116,13 +142,49 @@ public RoleBasedAuthorizationStrategy(Map<String, RoleMap> grantedRoles) {
       map = grantedRoles.get(PROJECT);
       itemRoles = map == null ? new RoleMap() : map;
   }
+  public static IdStrategy getSecurityRealmUserIdStrategy() {
+      final SecurityRealm securityRealm = Jenkins.getActiveInstance().getSecurityRealm();
+      IdStrategy userIdStrategy = securityRealm.getUserIdStrategy();
+      LOGGER.info("securityRealm.getUserIdStrategy() "+ userIdStrategy.getClass().getName());
+      return userIdStrategy;
+  }
+  public IdStrategy getUserIdStrategy() {
+    LOGGER.info("userIdStrategyChoice: "+ userIdStrategyChoice);
+    switch (userIdStrategyChoice) {
+      case CASE_INSENSITIVE:
+    LOGGER.info("userIdStrategyChoice: CASE_INSENSITIVE");
+        return new IdStrategy.CaseInsensitive();
+      case CASE_SENSITIVE:
+    LOGGER.info("userIdStrategyChoice: CASE_SENSITIVE");
+        return new IdStrategy.CaseSensitive();
+      case DEFAULT:
+      default:
+    LOGGER.info("userIdStrategyChoice: default: ");
+        return getSecurityRealmUserIdStrategy();
+    }
+  }
+  @DataBoundSetter
+  public void setUserIdStrategyChoice(UserIdStrategyChoice userIdStrategyChoice) {
+      LOGGER.info("setUserIdStrategyChoice: "+ userIdStrategyChoice);
+      this.userIdStrategyChoice = userIdStrategyChoice;
+      globalRoles.invalidateCache();
+      agentRoles.invalidateCache();
+      itemRoles.invalidateCache();
+  }
+  public void setUserIdStrategyChoiceStr(String userIdStrategyChoice) {
```

Comment:
I suggest not making such public method. 

```suggestion
  @Restricted(NoExternalUse.class)
  public void setUserIdStrategyChoiceStr(String userIdStrategyChoice) {
```